engine/runtime/src/main/scala/org/enso/compiler/core/IR.scala

```diff
@@ -7697,12 +7695,7 @@ object IR {
 
       /** @inheritdoc */
       @annotation.nowarn
-      override val location: Option[IdentifiedLocation] =
-        at match {
-          case ast: AST                => ast.location.map(IdentifiedLocation(_, ast.id))
-          case loc: IdentifiedLocation => Some(loc)
-          case _                       => None
-        }
+      override val location: Option[IdentifiedLocation] = at
```

Comment:
Returning `Option` is fine - good enough JVM compiler can eliminate `Option` overhead by proper escape analysis. The problem is storing `Option` in long time existing data structures. E.g. keep the original method and convert `null` to `None` and non-null to `Some(loc)`.

Reply:
I believe changing from `None` to `null` is beyond the scope of this PR.

Reply:
Please note that it is this PR which changes `null` to `Option`! All I am asking for is to avoid regression in memory usage by using `Option` where it wasn't used.